connector/src/main/scala/com/microsoft/kusto/spark/datasink/CountingCsvWriter.scala

```diff
@@ -33,13 +33,40 @@ case class CountingCsvWriter(out: Writer) {
 
       bytesCounter += 2
 
-      for (c <- str) {
-        if (c == '"') {
-          out.write("\"\"")
-          bytesCounter += 1
+      // Json does not allow special characters
+      if(nested){
```

Comment:
> Json does not allow special chars

I'm not sure I follow the logic. this function is called writeStringField, and is in a class called CsvWriter , 
why would you get a json?

Maybe we should handle this bad state else where?

Reply:
Kusto dynamic is json

Reply:
This is the writeStringField method in class CountingCsvWriter class.
This has absolutely nothing to do with Json.

If it needs to enclose the field in quotes, it should.
If it needs to double the escaping sequences, it should.

You can tale a look at how the C# CsvWriter works and handles JSON string fields for reference.

Reply:
I did 
This more resambles our JsonEscapedCsvWriter besides my addition of \f and \b
Do you suggest that the user will have to deal with it?